src/cpp/shared_core/FilePath.cpp

```diff
@@ -1198,16 +1198,12 @@ bool FilePath::isDirectory() const
 {
    try
    {
-      if (!exists())
-         return false;
-      else
-      {
-         return boost::filesystem::is_directory(m_impl->Path)
+      boost::system::error_code ec;
+      return boost::filesystem::is_directory(m_impl->Path, ec);
```

Comment:
I think the trailing semicolon here will cause an error with the following `|| isJunction()` on Windows. (I wonder if the Boost implementation already supports the detection of junctioned directories.)

Reply:
Boost release notes (1.57, long time ago) say: "The Windows implementation now treats NTFS directory junctions (also known as junctions, also known as mount points) as symlinks. This has the effect of treating directory junctions as directories, and thus supporting all operations suitable for directories."

I'll tweak the PR to ensure it builds on Windows and do a test to double-check that is_directory does indeed treat junctions as directories, and report back.

Reply:
Confirmed that Boost's `is_directory` does treat junction points as directories.